File: frontend/src/services/customerService.js
import api from './api';

const CUSTOMERS_URL = '/customers';

const customerService = {
  getAllCustomers: async () => {
    const response = await api.get(CUSTOMERS_URL);
    return response.data;
  },

  getCustomerById: async (id) => {
    const response = await api.get(`${CUSTOMERS_URL}/${id}`);
    return response.data;
  },

  createCustomer: async (customerData) => {
    const response = await api.post(CUSTOMERS_URL, customerData);
    return response.data;
  },

  updateCustomer: async (id, customerData) => {
    const response = await api.put(`${CUSTOMERS_URL}/${id}`, customerData);
    return response.data;
  },

  deleteCustomer: async (id) => {
    const response = await api.delete(`${CUSTOMERS_URL}/${id}`);
    return response.data;
  },

  getCustomersByStatus: async (status) => {
    const response = await api.get(`${CUSTOMERS_URL}/status/${status}`);
    return response.data;
  },

  getCustomersByCompany: async (company) => {
    const response = await api.get(`${CUSTOMERS_URL}/company/${company}`);
    return response.data;
  },
};

export default customerService; 

File: backend/customer-service/src/main/java/com/crmsystem/customer/service/CustomerService.java
package com.crmsystem.customer.service;

import com.crmsystem.customer.dto.CustomerDTO;
import com.crmsystem.customer.model.Customer.CustomerStatus;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.Map;

public interface CustomerService {
    
    CustomerDTO createCustomer(CustomerDTO customerDTO);
    
    CustomerDTO getCustomerById(Long id);
    
    CustomerDTO updateCustomer(Long id, CustomerDTO customerDTO);
    
    void deleteCustomer(Long id);
    
    Page<CustomerDTO> getAllCustomers(Pageable pageable);
    
    List<CustomerDTO> getCustomersByStatus(CustomerStatus status);
    
    Page<CustomerDTO> getCustomersByAssignedUser(Long userId, Pageable pageable);
    
    Page<CustomerDTO> searchCustomers(String searchTerm, Pageable pageable);
    
    boolean existsByEmail(String email);
    
    Map<CustomerStatus, Long> getCustomerStatusCounts();
} 